ui/src/client/ClientStore.ts

```diff
@@ -19,6 +19,13 @@ export class ClientStore extends BaseStore<IClient> {
             .then(() => this.snack('Client deleted'));
     }
 
+    @action
+    public update(id: number, name: string): Promise<void> {
+        return axios
+            .put(`${config.get('url')}client/${id}`, {name})
+            .then(() => this.snack('Client updated'));
```

Comment:
https://github.com/gotify/server/blob/178c76f410b69f7daaa8442ea051c105856cac74/ui/src/application/AppStore.ts#L38-L43

Please do a refresh after the update. Otherwise, the client name indicated on the clients list will not be updated on the list until the next page refresh.